### PR TITLE
Isolate recaptcha logic

### LINKED ISSUE #494: Isolate reCaptcha logic

### PR DESCRIPTION
**Pull request by ThaisMap**
We Isolated recaptcha validation in a separate component and a separate endpoint and removed that logic from claim create and claim review form

Closes #494

### CODE REVIEW

File: src/components/Claim/ClaimCreate.tsx [hunk 1]
```diff
@@ -122,20 +107,10 @@ const ClaimCreate = ({ personality, claim = {}, sitekey, edit = false }) => {
...
-            setDisableSubmit(true);
-            resolve();
-        });
-    }
 
-    const onChangeCaptcha = () => {
-        const recaptchaString = recaptchaRef.current.getValue();
-        setRecaptcha(recaptchaString);
+    const onChangeCaptcha = (captchaValid) => {
+        setDisableSubmit(!captchaValid)
```
Comment (thesocialdev) [inline src/components/Claim/ClaimCreate.tsx:112]: I'm wondering if the is going to be secure. If we do not check during the claim submit we open a vulnerability for the claim to be created by simply sending a Boolean value in the request. I think we can separate the logic but not create a new endpoint for that.

**Review by thesocialdev — CHANGES_REQUESTED** [review]

File: src/components/Claim/ClaimCreate.tsx (hunk 1 above)
Comment (thesocialdev) [reply src/components/Claim/ClaimCreate.tsx:112]: @pepermao can help with that if needed, but I think the best solution would be to create a [ValidationPipe](https://docs.nestjs.com/techniques/validation#using-the-built-in-validationpipe) for the DTO that does the recaptcha check.

Comment (ThaisMap) [reply src/components/Claim/ClaimCreate.tsx]: Considering the problems we had with injecting configService in a validationPipe, see if we can use the captcha service approach that I tried on next commit.

**Comment by thesocialdev** [comment]
@ThaisMap we need to rebase it now, but I'm ready to approve it

**Review by thesocialdev — APPROVED** [review]

**Pull request merged**